### PR TITLE
python@3.9 3.9.7

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

File: Formula/python@3.9.rb [hunk 1]
```diff
@@ -57,19 +57,20 @@ class PythonAT39 < Formula
   link_overwrite "Frameworks/Python.framework/Resources"
   link_overwrite "Frameworks/Python.framework/Versions/Current"
 
+  # Find bundled versions at https://github.com/python/cpython/blob/v#{version}/Lib/ensurepip/__init__.py
```
Comment (branchvincent) [inline Formula/python@3.9.rb:60]: is this comment accurate? asking because for 3.9.6, cpython shipped with setuptools 56.0.0 while we used 57.0.0

Comment (carlocab) [reply Formula/python@3.9.rb:60]: I think we always just ship the latest versions here.

Comment (Bo98) [reply Formula/python@3.9.rb:60]: Yes, latest. Lines 284-287 overwrite the versions in this file.

Comment (branchvincent) [reply Formula/python@3.9.rb]: thanks for the clarification, updated the comment

**Comment by iMichka** [comment]
The `cake` bottle is an `:all` bottle but it depends on `mono` which has no ARM bottle:
```
2021-08-31T03:23:45.6587770Z [34m==>[0m [1mbrew install --only-dependencies cake[0m
2021-08-31T03:23:46.2981680Z [31m==>[0m [1m[31mFAILED[0m[0m
2021-08-31T03:23:46.2982480Z [31mError:[0m mono: no bottle available!
2021-08-31T03:23:46.2982930Z You can try to install from source with:
2021-08-31T03:23:46.2983560Z   brew install --build-from-source mono
2021-08-31T03:23:46.2984170Z Please note building from source is unsupported. You will encounter build
2021-08-31T03:23:46.2985250Z failures with some formulae. If you experience any issues please create pull
2021-08-31T03:23:46.2986230Z requests instead of asking for help on Homebrew's GitHub, Twitter or any other
2021-08-31T03:23:46.2986720Z official channels.
```

**Comment by iMichka** [comment]
Fix for `copilot`: #84352

**Comment by branchvincent** [comment]
ARM
- [x] `cake`:  unbottled dep
- [ ] `copilot`: #84352
- [ ] `eureka`
- [ ] `fauna-shell`
- [ ] `lean`
- [ ] `ledger`
- [ ] `mathlibtools`
- [ ] `notcurses`
- [ ] `oakc`
- [ ] `opensearch-dashboards`
- [ ] `ponysay`

**Comment by carlocab** [comment]
I think `ponysay` is pre-existing. Seen also at #84157.

**Comment by carlocab** [comment]
`lean` and `mathlibtools` are a test-bot bug. If you look earlier in the CI log you'll see a `brew unlink coreutils` without a corresponding `brew link coreutils`, which is why neither formula can find `greadlink`.

**Comment by carlocab** [comment]
I think testing lots of Java formulae on Linux is just weird. Or at least, that's what I remember @iMichka saying. Input from @Homebrew/linux appreciated.

**Comment by SMillerDev** [comment]
Swift has a rework of the formula waiting. Any issues we can fix there after merging this.

**Comment by iMichka** [comment]
Regarding the java formulae: we have already seen the errors before, I think it's due to a mix of openjdk 11 and openjdk, or maybe something else, but it only happens on test-bot. Feel free to ignore these for now. If someone wants to investigate, go ahead, I won't have time before a few days to focus on this, as it looks like a bigger issue and I have no clue about the java ecosystem.

**Comment by carlocab** [comment]
Swift (#73984) looks ready to merge to me, actually

**Comment by branchvincent** [comment]
can't reproduce `ledger` on big sur after downloading the bottle

**Comment by SMillerDev** [comment]
Do we want to retry the build?

**Comment by carlocab** [comment]
Restarted.

**Comment by cclauss** [comment]
Tests 10.15 are still running but results so far are:

brew command | 11-arm64 | 11 | 10.15 | 10.14
--- | --- | --- | --- | ---
install --only-dependencies cake | FAIL | | |
install cake | FAIL | | |
test --retry --verbose copilot | FAIL | FAIL | | FAIL
test --retry --verbose eureka | FAIL | FAIL | | FAIL
test --retry --verbose fauna-shell | FAIL | FAIL | | FAIL
install --only-dependencies gmsh | | | | FAIL
install gmsh | | | | FAIL
test --retry --verbose hyperkit | | FAIL | | FAIL
test --retry --verbose lean | FAIL | FAIL | | FAIL
test --retry --verbose llvm@7 | | FAIL | |
test --retry --verbose mathlibtools | FAIL | FAIL | | FAIL
test --retry --verbose notcurses | FAIL | FAIL | | FAIL
install --only-dependencies opensearch-dashboards | FAIL | | |
install opensearch-dashboards | FAIL | | |
brew test --retry --verbose swift | | | | FAIL

**Comment by iMichka** [comment]
I think @carlocab forgot to rebase, so the fixes that have been made were not taken into account. Either we rebase and build again, or we just merge this as-is.

**Comment by carlocab** [comment]
Formulae like `eureka` shouldn't be failing again, though, since the fix is in `brew` and not the formula.

**Comment by carlocab** [comment]
Let's rebase just to be sure...

**Comment by cclauss** [comment]
Other tests are still running but results so far are:

brew command | 11-arm64 | 11 | 10.15 | 10.14
--- | --- | --- | --- | ---
install --only-dependencies cake | FAIL | | |
install cake | FAIL | | |
test --retry --verbose notcurses | FAIL | | |
install --only-dependencies opensearch-dashboards | FAIL | | |
install opensearch-dashboards | FAIL | | |

**Comment by iMichka** [comment]
`wownero` has been fixed in  #84624

**Comment by iMichka** [comment]
`cake` was fixed in #84630

**Comment by carlocab** [comment]
llvm@7: https://github.com/Homebrew/homebrew-core/pull/84649

**Comment by carlocab** [comment]
`gmsh` depends on `opencascade`, which has no Mojave bottle.

**Comment by cho-m** [comment]
macOS looks fine to me.

On Linux, outside this PR, we really need to get rid of the OpenJDK CI problem.
Ignoring that, the remaining include:
- `buku`
- `streamlink` - 403 Client Error: Forbidden for url: https://vimeo.com/189776460. If unable to reproduce outside CI, then it may be Google Cloud blocking it.
- `cabal-install` - curl linkage issue
- `ninja` - curl linkage issue

Fixes:
- `mpi4py` - #84623
- `scrollkeeper` - #84622

Same errors as my Perl PR, so probably unrelated to Python:
- `julia`
- `kafka`
- `mrbayes`
- `parquet-cli` - maybe related to Java CI issue

---

We could pursue most outside this PR.

Only one I would recommend checking is `buku`, which passed for me in Perl PR https://github.com/Homebrew/homebrew-core/runs/3502105423?check_suite_focus=true#step:10:4599

**Review by carlocab — APPROVED** [review]

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Let's ship this. I'm going to have a look at the curl/ninja linking issue, because it is more concerning. Regarding the java issues on Linux it's on my long term to-do list but I do not plan to work on this in the next weeks.

**Pull request closed without merging**